app/scripts/views/tour/tour-page-controller.js

```diff
@@ -42,6 +47,15 @@
 
             ctl.onPrevClick = onPrevClick;
             ctl.onNextClick = onNextClick;
+
+            Visualization.get(Config.ids.tour).then(onVisReady, onVisError);
```

Comment:
I understand wanting a separate manager for the visualizations, and separating controller and page logic, but this seems to be complicating matters, particularly with throwing an error for an expected condition in the visualization `get`.

An alternative suggestion:  maybe pass the viz ready/error callbacks to `add`, let `add` create the viz and invoke the callbacks when creation is `done`.  Don't add the viz ID to the collection until creation is done, and let `get` be a no-op until it's ready.